### DISCUSSION TITLE
SvelteKit integration

### DISCUSSION THREAD

**Discussion by the1323** [bergie/passport-saml#915] [Q&A]
Has anyone able to integrate passport saml to SvelteKit application?
is it even possible?

**Comment by markstos** [comment]
SvelteKit is a frontend framework, while password-saml and node-saml are backend libraries. 

Yes, some integration would be possible, just frontend frameworks can be made to call other backend APIs. You could read and follow the passport-saml and node-saml docs here and integrate with it as you would other backend code.

**Comment by markstos** [comment]
If you are asking if you can create a solution that's completely on the frontend, then I don't think that will work. 

If you want to validate signatures against your private key, there's no way to secure your keys if they are stored on the frontend.
